### ISSUE TITLE
Implement Iron List

### ISSUE THREAD

**Issue by Joltrix** [StarcounterApps/Timeline#25]
Currently, we display X events, and we have a "load more events" button.

Once this button is clicked, a patch is sent from the server containing all the new events. Which smoothly updates the view rather than loading all events (which could theoretically be 1000 events, or 100.000 events) at the same time.

Another approach to this would be to use Iron list, or something similar. Thus rendering only the "visible" events, rather than rendering all of them.

https://www.webcomponents.org/element/PolymerElements/iron-list

But perhaps this can be postponed to a later version of the Timeline app

**Comment by andwah** [comment]
At the moment I rather see a "Load more" button then an `infinite` list. 

Imagine that we have 3 columns and one app in each column and all of them are using an  `infinite` list.  

Wonder if the scrolling/loading will work "out of the box"  and feel natural for the enduser.

**Comment by Joltrix** [comment]
Ah okay, then we can leave this issue for now, rethink it, and implement it if we want it later.

What if we try to somehow try combine it together with the load more button?

Because the goal is to try to reduce the lag. Now that @alshakero has fixed the palindrom lag, only the rendering lag remains.

**Comment by andwah** [comment]
Can we see what is lagging?  is it <starcounter-include>, or code that "we" have done, if so is it fixable? 

if it's out of our hands then we may try to do some workarounds to solve the lagging

**Comment by alshakero** [comment]
One way to fix half of the rendering lag is to calculate the height needed for the upcoming to-be-appended notes, extend the div's height to it by JS, then append the notes. This will be much faster.

E.g: note's height is 150 + 20 px (20 is margins). Coming notes are 50. 5 per row, then height is 10 notes = 1700 px. Change div's height to that and you'll see much better perf. 

This happens because changing the height of the viewport causes a reflow, which is a very expensive task. So it's smarter to reflow only once.
